test/com/google/javascript/jscomp/Es6VariableReferenceCheckTest.java

```diff
@@ -277,6 +277,10 @@ public void testClass() {
     assertNoWarning("class A { f() { return 1729; } }");
   }
 
+  public void testRedeclareClassName() {
+    assertNoWarning("var Clazz = class Foo {}; var Foo = 3;");
```

Comment:
I think the ES6 specs was trying to be consistent with how `function`s look like, i.e. `var func = function f() {}`.

@jleyba Do you have a real-world example showing the usage of this?